urla05239b71c_viniciusinnovatium_mumps2java_api_tgz-pJ8-com_disclinc_netmanager_script_test_ExecTesteNewJ8/src/br/com/innovatium/mumps2java/dataaccess/DAO.java

```diff
@@ -0,0 +1,216 @@
+package br.com.innovatium.mumps2java.dataaccess;
+
+import java.sql.Connection;
+import java.sql.DatabaseMetaData;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.HashMap;
+import java.util.Map;
+
+import br.com.innovatium.mumps2java.todo.TODO;
+
+public class DAO {
+	private final Connection con;
+
+	public DAO() {
+		this(ConnectionType.DATASOURCE);
+	}
+
+	public DAO(ConnectionType connectionType) {
+		try {
+			con = ConnectionFactory.getConnection(connectionType);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code assigns a resource to a class field and there is no method that closes or returns this resource. This can result in a resource leak that slows down or crashes your system.

**Fix**
Create a method in this class that closes the following resource: *con*. After you use this resource, use this method to prevent a resource leak.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).